rse-narrative.md

```diff
@@ -0,0 +1,157 @@
+
+### Chapter 3: Going further with the unix shell
+
+The narrative in this chapter currently revolves around a little bash script called `years.sh`,
+which extracts a list of years from a climate data file.
+It will need to be replaced with a bash script
+that extracts basic information from the ebook text files in `zipfs-law/data/`.
+
+To be more specific, a new bash script `book_summary.sh` could extract the
+book title, author and release date using the `head` command to select the relevant lines
+from an ebook text file. 
+(Some of the original text files will need to be edited very slightly
+to ensure that this information is on exactly the same line in each file.)
+Later on in the chapter,
+the content on `grep` could then show how it could have been used to locate the same
+information in the files 
+(i.e. so the information wouldn't need to be on exactly the same line in each file.)
+
+### Chapter 4: Command line programs
+
+This chapter currently involves a generic script called `script_template.py`,
+which can be configured using command line arguments or a configuration file.
+Instead of developing the generic `script_template.py`,
+this chapter could develop `countwords.py` instead.
+(The current version of `countwords.py` will need to be updated to use argparse
+and other libraries/concepts introduced in that chapter.)
+
+### Chapter 5: Git at the command line
+
+The narrative in this chapter currently involves Frances and her colleague Jean Jennings
+writing a history of the ENIAC project.
+Instead, in this chapter the `collate.py` script could be written from scratch,
+tracking changes with Git along the way.
+
+### Chapter 6: Advanced Git
+
+This chapter introduces the concept of branches.
+In the current draft, a violin plot is created on one branch
+and a beeswarm plot on the other.
+The better looking plot is then merged into the master branch.
+This example could be tweaked so that two slightly different versions of 
+`plotcounts.py` are developed instead
+(i.e. two different ways of plotting the word count data).
+
+### Chapter 7: Code style, review and refactor
+
+It's not clear how much of this chapter will ultimately end up in the novice Python book,
+but nonetheless code review will definitely be introduced here.
+The chapter text could be edited such that a draft version of `testfit.py`
+(with deliberate flaws written in)
+is presented and then reviewed/improved.
+
+### Chapter 8: Automating analyses 
+
+This chapter can stay pretty much as is.
+The introductory paragraphs about Zipf's Law can be removed 
+(because they should now appear in the introduction chapter)
+and the workflows developed in the chapter could be extended to also include
+`plotcounts.py` and `testfit.py`.
+
+### Chapter 10: Project structure
+
+Now that we've written all our scripts for the Zipf's Law analysis,
+this chapter should structure it all properly (i.e. with license files and README's etc).
```

Comment:
This I think for the R side would be better at the beginning, to begin structuring everything in an R package format.

Reply:
I agree that the R book could have a completely different order for the chapters - I've edited the document so it only makes reference to the Python book.